x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackTemplateRegistry.java

```diff
@@ -253,4 +258,13 @@ protected boolean requiresMasterNode() {
         // there and the ActionNotFoundTransportException errors are then prevented.
         return true;
     }
+
+    @Override
+    protected boolean isClusterReady(ClusterChangedEvent event) {
```

Comment:
@eyalkoren Now checking all nodes are at least at v 8.9.0, so it does not fails because of the ignore_missing_pipeline parameter

Reply:
What are you preventing with this?
AFAIK, once the local implementation of [`requiresMasterNode()`](https://github.com/elastic/elasticsearch/blob/6179ae84f0cda5e2cb9e8070bcd88ce5ae699b9d/x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackTemplateRegistry.java#LL247C31-L247C37) returns `true`, as in this case, this ensures that the upgrade will occur only on the elected master. I believe that during rolling upgrades this ensures that this happens only after all non-master nodes are already upgraded. 
Since the usage of `ignore_missing_pipeline` was introduced in https://github.com/elastic/elasticsearch/pull/95971, which was added to 8.9.0 and not back-ported, I am not sure whether this is required.

BTW, `AnalyticsTemplateRegistry` also [requires master node](https://github.com/elastic/elasticsearch/blob/6179ae84f0cda5e2cb9e8070bcd88ce5ae699b9d/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/AnalyticsTemplateRegistry.java#L147), so double-check if this is required in the original case as well.

@jbaiera @dakrone please confirm or enlighten me if I got it wrong

Reply:
There is no guarantee that the non-master nodes will be upgraded first during a rolling upgrade. I this is a sensible precaution.

Reply:
Would nodes apply settings coming from a master of a higher version?

Reply:
@dakrone just explained that I indeed got it wrong and there is no actual enforced guarantee as to the order of upgrade in code. This should normally not occur if the upgrade is done according to documentation (master node last), but such verification does make sense